spextra/spextra.py

```diff
@@ -397,33 +443,39 @@ def flat_spectrum(cls, amplitude=0, waves=None):
         spex : Spextrum
             New ``Spextrum`` instance.
         """
-        if waves is None:
-            waves = _default_waves()
-        waves = _angstrom_value(waves)
-        amplitude = _abmag_qty(amplitude)
+        waves = (waves or _default_waves()).to(u.AA).value
```

Comment:
This is broken, because waves is a Quantity, and "Quantity truthiness is ambiguous, especially for logarithmic units and temperatures. Use explicit comparisons". In particular, the example in https://github.com/AstarVienna/ScopeSim_Templates/blob/main/scopesim_templates/calibration/calibration.py#L44 raises the quoted ValueError. Let me take a look

Reply:
Fixed in https://github.com/AstarVienna/speXtra/pull/57 , with tests